src/Controller/ProductApiController.php

```diff
@@ -0,0 +1,133 @@
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class ProductApiController extends AbstractController
+{
+    /**
+     * @Route("/api", name="api", methods={"GET"})
+     */
+    public function index(
+        ProductInfoService $productHandler,
+        CsvParser $csvParser,
+        Request $request
+    ): Response {
+
+        $parsedProducts = $csvParser->getParsedProducts();
+
+        $absolutePath = rtrim($request->server->get('SYMFONY_DEFAULT_ROUTE_URL'), '/')
+            .$request->server->get('REQUEST_URI');
+
+        return new JsonResponse(
+            $productHandler->getProductsLinksList($parsedProducts, $absolutePath),
+            Response::HTTP_OK,
+            ['Content-Type: application/json']
+        );
+    }
+
+    /**
+     * @Route("/api/info", name="api_info", methods={"GET"})
+     */
+    public function info(
+        Request $request,
+        UserAgentFactory $factory,
+        UserAgentInfoVisitLogger $handler
+    ): Response {
+        $userAgent = $factory->create(
+            $request->getClientIp(),
+            $request->getPreferredLanguage(),
+            $request->server->get('HTTP_SEC_CH_UA')
+        );
+
+        $handler->writeUserAgentInfo($userAgent);
+
+        return new JsonResponse(
+            'This is info',
+            Response::HTTP_OK,
+            ['Content-Type: application/json']
+        );
+    }
+
+    /**
+     * @Route("/api/{slug}", name="api_product_show", defaults={"_format": "json"}, methods={"GET"})
+     */
+    public function show(
+        string $slug,
+        ProductInfoService $productHandler,
+        CsvParser $csvParser,
+        TranslatorInterface $translator
+    ): Response {
+        $parsedProducts = $csvParser->getParsedProducts();
+        $product = $productHandler->getProductInfo($slug, $parsedProducts);
+        $productTranslated = [];
+
+        foreach ($product as $key => $value) {
+            $translatedKey = $translator->trans($key);
+            $productTranslated[$translatedKey] = $value;
+        }
+
+        return new JsonResponse(
+            json_encode($productTranslated, 256),
+            Response::HTTP_OK,
+            ['Content-Type: application/json'],
+            true
+        );
+    }
+
+    /**
+     * @Route("/api/notify", name="api_notify", defaults={"_format": "json"}, methods={"POST"})
+     */
+    public function notify(
+        Request $request,
+        NotificationFactory $notificationFactory,
+        SendNotificationCommand $notificationCommand
+    ): Response {
+        try {
+            $data = $request->toArray();
+
+            $notification = $notificationFactory->create(
+                $data['recipient'],
+                $data['message'],
+                $data['channel']
+            );
+
+            $notificationCommand->send($notification);
+
+            $result = true;
+        } catch (\Throwable $exception) {
+            $result = false;
```

Comment:
it makes sense to write an error to the log

Reply:
It also makes sense to have a return statement inside the catch statement to simplify the readability of the method.